### PR TITLE
PKCE

### PR DESCRIPTION
**Pull request by anehx**

### CODE REVIEW

File: addon/routes/oidc-authentication.js [hunk 1]
```diff
@@ -31,6 +34,18 @@ export default class OIDCAuthenticationRoute extends Route {
...
+    if (this.config.enablePkce) {
+      let pkceCodeVerifier = sessionStorage.getItem("pkceCodeVerifier");
+
+      if (!pkceCodeVerifier) {
+        pkceCodeVerifier = generateCodeVerifier(96);
+        sessionStorage.setItem("pkceCodeVerifier", pkceCodeVerifier);
+      }
+
+      this.pkceCodeVerifier = pkceCodeVerifier;
+    }
```
Comment (anehx) [inline addon/routes/oidc-authentication.js:48]: For the `nextURL` we use the local storage (via session data) for the same reason. Let's keep it consistent and use the session as well. You can look at the implementation here: https://github.com/adfinis/ember-simple-auth-oidc/blob/80460c0019daab976d72912227d836e1d6adf3a1/addon/services/session.js#L67-L88

File: addon/routes/oidc-authentication.js [hunk 2]
```diff
@@ -97,10 +112,16 @@ export default class OIDCAuthenticationRoute extends Route {
...
 
-    await this.session.authenticate("authenticator:oidc", {
+    const data = {
       code,
       redirectUri: this.redirectUri,
-    });
+    };
+
+    if (this.config.enablePkce) {
+      data.codeVerifier = this.pkceCodeVerifier;
```
Comment (anehx) [inline addon/routes/oidc-authentication.js:121]: Let's take this from the session

File: addon/routes/oidc-authentication.js [hunk 3]
```diff
@@ -129,16 +150,22 @@ export default class OIDCAuthenticationRoute extends Route {
...
-    ]
-      .filter(Boolean)
-      .join("&");
+    ];
+
+    if (this.config.enablePkce) {
+      const pkceChallenge = generatePkceChallenge(this.pkceCodeVerifier);
+      search.push(`code_challenge=${pkceChallenge}`);
+      search.push("code_challenge_method=S256");
+    }
```
Comment (anehx) [inline addon/routes/oidc-authentication.js:166]: Same here

**Review by anehx — COMMENTED** [review]
Some minor changes needed, otherwise LGTM, good job! :100:

File: addon/routes/oidc-authentication.js (hunk 1 above)
Comment (Yelinz) [reply addon/routes/oidc-authentication.js]: Did not spot that, changed to use `session.data`

File: addon/routes/oidc-authentication.js (hunk 2 above)
Comment (Yelinz) [reply addon/routes/oidc-authentication.js]: Changed

File: addon/routes/oidc-authentication.js (hunk 3 above)
Comment (Yelinz) [reply addon/routes/oidc-authentication.js]: Changed

**Comment by anehx** [comment]
Nice, good job! :fire:

**Pull request merged**